source/adapters/cuda/command_buffer.cpp

```diff
@@ -862,11 +879,7 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferUpdateKernelLaunchExp(
     return UR_RESULT_ERROR_INVALID_OPERATION;
   }
 
-  if (auto NewWorkDim = pUpdateKernelLaunch->newWorkDim) {
-    // Error if work dim changes
-    if (NewWorkDim != hCommand->WorkDim) {
-      return UR_RESULT_ERROR_INVALID_OPERATION;
```

Comment:
we don't need to check for this error anymore?

Reply:
This has been changed due to the discussions in https://github.com/oneapi-src/unified-runtime/pull/1924#discussion_r1754031968 . The check is now for when the workdim is changed, the global work size and global work offset are also updated.